CONTRIBUTING.md

```diff
@@ -296,6 +296,89 @@ OS and is built by Android NDK. This often introduces compatibility issues as
+
+[Fixes https://github.com/termux/termux-packages/issues/<issue number>]
+[Closes https://github.com/termux/termux-packages/pull_requests/<pr number>]
+```
+
+- `<commitType>` describes the type of commit. Types of commits:
+  - `upgpkg`: One or more package(s) was/were updated.
+    Optional extended commit message may include new list of features in the new version, and detailed list of changes in build scripts and/or patches
+  - `dwnpkg`: One or more package(s) was/were downgraded due to build issues or potential bugs
+    commit message should include summary of why the package(s) was/were downgraded. If the summary can't completely describe the reason for downgrade, extended commit message should contain full reason for downgrade.
+  - `enhance`: Enable a feature in a package that was earlier not enabled.
+    Optional (but highly recommended) extended commit message may contain detailed summary of the enabled feature(s) and a basic use case
+  - `chore`: Any housekeeping change or a change which does not affect the user in any way.
+  - `rebuild`: Rebuild a package to link against newer version of shared library
+    Special cases:
+    - When mass rebuilding packages depending on a major package (e.g. openssl), consider using this format:
+      ```
+      rebuild(deps:main/openssl): link against OpenSSL 3.0
+      ```
+  - `scripts`: Any change affecting our build scripts
+
+Examples of good commit messages:
+1. ```
+   upgpkg(main/nodejs): upgrade to v18.2.0
+   ```
+
+2. ```
+   dwnpkg(main/htop): downgrade to v2.2.0
+
+   v3.x needs access to /proc/stat which is now restricted by Android
+   ```
+
+3. ```
+   enhance,upgpkg(nodejs): update to v18.2.0 and use shared libuv
+
+   # Describe the technical reasons of how using shared libuv is beneficial
+   ```
+
+### Special notes for newbies who're just getting started with Open Source
+
+In order to encourage new contributors and help them conteibute to open source, the above mentioned commit requirements should be optionally relaxed. In cases where commit messages need to be changed, the PR may be **Squashed and Merged** or may be merged manually from the command line.
+
+#### Notes for merging PRs from command line
+
+1. It is recommended to use the [GitHub CLI (`gh`)](https://cli.github.com) in order to fetch the contributor's branch.
+   ```sh
+   gh pr checkout <PR Number>
+   ```
+
+2. After checking out the branch, ammend the commit message and optionally rebase against the master branch (if necessary).
+
+   When merging manually make sure that you give proper credits for the original patch to it's author by adding a `Co-authored-by: ` line. See https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors for more details. Also add a `Closes #<PR number>`
+
+   ```sh
+   git commit --amend # Will open up your editor to ammend the commit message
```

Comment:
ammend -> amend

Reply:
Makes more sense to have `git commit --amend` after `git rebase origin/master`. If the amended commit is to be signed (`git commit --amend -S`) it otherwise have to be signed twice